src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -975,6 +980,17 @@ def set_up_runtime(self):
                         "java": runtime_java_def
                     }
             safe_set(self.containerapp_def, "properties", "configuration", "runtime", value=runtime_def)
+    
+    # pylint: disable=unsupported-assignment-operation
+    def set_up_scale_rule(self):
+        scale_def = super().set_up_scale_rule()
+        if scale_def and scale_def["rules"] and scale_def["rules"][0] and scale_def["rules"][0]["custom"] and self.get_argument_scale_rule_identity():
+            identity = self.get_argument_scale_rule_identity().lower()
+            if identity != "system":
+                subscription_id = get_subscription_id(self.cmd.cli_ctx)
```

Comment:
What if the identity resourceId not in current subscription?
The `_ensure_identity_resource_id` change the subscription Id to current sub, which is not expect.
![image](https://github.com/Azure/azure-cli-extensions/assets/46143499/188de8cf-608f-467e-a018-2274d5da61da)